### PR TITLE
Fix several different types of govet warnings in our code

### PR DESCRIPTION
**Pull request by Martin2112**

### CODE REVIEW

File: merkle/memory_merkle_tree_test.go [hunk 1]
```diff
@@ -528,8 +528,8 @@ func TestMerkleTreePathBuildOnce(t *testing.T) {
 		}
 
 		for j := 0; j < len(p2); j++ {
-			if bytes.Compare(p1[j].Value.hash, decodeHexStringOrPanic(testPaths[i].testVector[j])) != 0 {
-				t.Errorf("Path mismatch: %s %s")
+			if got, want := p1[j].Value.hash, decodeHexStringOrPanic(testPaths[i].testVector[j]); bytes.Compare(got, want) != 0 {
+				t.Errorf("Path mismatch: got: %v want: %v", p1[j].Value.hash, )
```
Comment (daviddrysdale) [inline merkle/memory_merkle_tree_test.go]: Parameter missing, should probably be ", got, want)"

File: examples/ct/ct_handlers.go [hunk 2]
```diff
@@ -251,9 +251,9 @@ func addChainInternal(w http.ResponseWriter, r *http.Request, c RequestHandlers,
 
 	request := trillian.QueueLeavesRequest{LogId: c.logID, Leaves: []*trillian.LeafProto{&leafProto}}
 
-	ctx, _ := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
-
+	ctx, cancelFunc := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
 	response, err := c.rpcClient.QueueLeaves(ctx, &request)
+	cancelFunc()
```
Comment (daviddrysdale) [inline examples/ct/ct_handlers.go]: Not sure what the standard paradigm is here, but various examples seem to prefer "defer cancelFunc()" a line higher - thoughts?

File: merkle/memory_merkle_tree_test.go (hunk 1 above)
Comment (Martin2112) [inline merkle/memory_merkle_tree_test.go]: Oops, fixed.

File: examples/ct/ct_handlers.go (hunk 2 above)
Comment (Martin2112) [inline examples/ct/ct_handlers.go]: The docs say you should call cancel as soon as the work is complete but deferring it guarantees it gets called regardless of what happens afterwards. I think I'll go with defer.

**Comment by Martin2112** [comment]
We can look at turning govet back on. With the previous mocking framework there were warnings it was impossible to get rid of, that's probably no longer the case.

**Pull request merged**